integration-test/src/main/java/org/apache/iotdb/it/env/AbstractEnv.java

```diff
@@ -200,14 +193,53 @@ public void cleanAfterTest() {
     cleanupEnvironment();
   }
 
-  @Override
-  public Connection getConnection() throws SQLException {
-    return new ClusterTestConnection(getWriteConnection(null), getReadConnections(null));
+  public final int[] searchAvailablePorts() {
+    do {
+      int randomPortStart = 1000 + (int) (Math.random() * (1999 - 1000));
+      randomPortStart = randomPortStart * 10 + 1;
+      File lockFile = new File(getLockFilePath(randomPortStart));
+      if (lockFile.exists()) {
+        continue;
+      }
+
+      List<Integer> requiredPorts =
+          IntStream.rangeClosed(randomPortStart, randomPortStart + 9)
+              .boxed()
+              .collect(Collectors.toList());
+      String cmd = getSearchAvailablePortCmd(requiredPorts);
+
+      try {
+        Process proc = Runtime.getRuntime().exec(cmd);
+        if (proc.waitFor() == 1 && lockFile.createNewFile()) {
+          return requiredPorts.stream().mapToInt(Integer::intValue).toArray();
+        }
+      } catch (IOException | InterruptedException ex) {
+        // ignore
```

Comment:
Is  add "Thread.currentThread().interrupt();"   ?

Reply:
Good catch! It has been fixed.